### DISCUSSION TITLE
Pattern  across extended timelines

### DISCUSSION THREAD

**Discussion by Samraw003** [TDAmeritrade/stumpy#396] [Q&A] [answered]
Hi Sean,

As I shared earlier I have BEA data on Personal Consumption Expenditure.  I ran stumpy on one series with date range from 2000 t0 2010 and the following was found:

> The motif is located at index 70 The nearest neighbor is located at index 19

When I extended the timelines: 2000 to 2020 (additional 10 years) the following was found:

> The motif is located at index 237 The nearest neighbor is located at index 231

The question is: Why earlier motif was not found when the data existed?

Regards

Sammeer

**Comment by seanlaw** [comment, marked as answer]
@Samraw003 This suggests that as you added more data, the global minima (motif) has changed. I recommend that you plot the matrix profile and see how it changes as you add one more year at a time. What you'll likely see is that the global minima will drop lower as one particular year gets added. 

Having said that, the original motif (located at index 70) may still be there if plot the data. It's just not the global motif (i.e., the one that has the smallest distance to its nearest neighbor).

**Comment by Samraw003** [comment]
@seanlaw . Thanks for the prompt clarifications.

This makes sense as with the current code (your example) we are just picking the lowest values of distance.

In one of your videos, you demoed a code that displayed all the motifs on button click. Will it be possible to get _that_ piece of code?

Regards

Sammeer

**Reply by seanlaw** [reply]
I don't recommend that visualization because it doesn't actually give you the motifs. All it does is takes a precomputed matrix profile and shows you the nearest neighbor for every subsequence in your time series. Additionally, the demo is hardcoded for that specific dataset and I am unable to provide support for it. In the next release, we will likely provide some additional tools for motif discovery.

**Reply by Samraw003** [reply]
@seanlaw Great!! Thanks a ton!
